src/circuit/gadget/ecc/chip/add.rs

```diff
@@ -407,22 +407,50 @@ pub mod tests {
         assert_ne!(p_val, q_val);
 
         // Check complete addition P + (-P)
-        p.add(layouter.namespace(|| "P + (-P)"), p_neg)?;
+        {
+            let sum = p.add(layouter.namespace(|| "P + (-P)"), p_neg)?;
+            assert_eq!(&sum, zero);
+        }
 
         // Check complete addition 𝒪 + 𝒪
-        zero.add(layouter.namespace(|| "𝒪 + 𝒪"), zero)?;
+        {
+            let sum = zero.add(layouter.namespace(|| "𝒪 + 𝒪"), zero)?;
+            assert_eq!(&sum, zero);
+        }
 
         // Check P + Q
-        p.add(layouter.namespace(|| "P + Q"), q)?;
+        {
+            let sum = p.add(layouter.namespace(|| "P + Q"), q)?;
+            let real_sum = Point::new(
+                chip.clone(),
+                layouter.namespace(|| "endo(P)"),
+                Some((p_val + q_val).to_affine()),
+            )?;
+            assert_eq!(sum, real_sum)
```

Comment:
Here, we are allocating both the sum and the expected result in the circuit, and then checking equality outside the circuit.